main.cc

```diff
@@ -2689,5 +2725,18 @@ void change_club(int new_club_idx, int player) {
 	strncpy(gameb.club[new_club_idx].manager, gameb.club[old_club_idx].manager, 16);
 	strncpy(gameb.club[old_club_idx].manager, DEFAULT_MANAGER_NAME, 16);
 }
+
+void dump_free_players() {
+	printf("CLUB NAME	    PLAYER NAME	HN TK PS SH HD CR FT F M A AG  WAGES\n");
+	for (int i = 0; i < 244; ++i) {
+		struct gameb::club &club = get_club(i);
+		for (int i = 0; i < 24; ++i) {
+			struct gamec::player &p = gamec.player[ club.player_index[i] ];
+			if (club.league != 0 && p.hn < 100 && p.contract == 0) {
+				dump_player_row(p, club);
+			}
+		}
+	}
+}
```

Comment:
This feels a bit off, why wouldn't you just loop over the gamec data directly?

Also, why are you checking for hn < 100?

Reply:
Two reasons for the loops that way, but open to suggestions:
- I wanted to group the results by club, this seemed to be the easiest way so I could print out each row as I found them, without creating a data structure to hold them.
-  I couldn't see the club in gamec, or a way to look up the player's club in gameb.

The `hn < 100` check is because when looping the 24 players, if the team has less than 24 players, it's just garbage data, and the `hn` was always 255. So this is essentially checking if the row has a valid player in it. Is there a cleaner way?

Reply:
Oh, I see what you're doing now. You're looking for players that aren't on the transfer-market, but can be bought by phoning the club directly.

Modify the inner for-loop to
```
for (int i = 0; i < 24; ++i) {
  if (club.player_index[i] == -1)
    continue; // skip this empty slot

  ...rest of code...
}
```
As if a club does not have all 24 players, the slot for a playerindex is -1.